File: app/src/main/java/org/worshipsongs/service/ResetDefaultSettingsService.java
package org.worshipsongs.service;

import android.content.Context;
import android.content.DialogInterface;
import android.content.SharedPreferences;
import android.preference.DialogPreference;
import android.preference.PreferenceManager;
import android.util.AttributeSet;

import org.worshipsongs.R;

/**
 * Author : Seenivasan, Madasamy
 * Version : 1.x
 */
public class ResetDefaultSettingsService extends DialogPreference
{
    protected Context context;

    public ResetDefaultSettingsService(Context context, AttributeSet attrs)
    {
        super(context, attrs);
        this.context = context;
        setTitle(R.string.reset_default_title);
    }

    @Override
    public void onClick(DialogInterface dialog, int which)
    {
        super.onClick(dialog, which);
        if(which == DialogInterface.BUTTON_POSITIVE)
        {
            SharedPreferences.Editor preferencesEditor = PreferenceManager.getDefaultSharedPreferences(this.context).edit();
            preferencesEditor.clear();
            PreferenceManager.setDefaultValues(context, R.xml.settings, true);
            preferencesEditor.apply();
            getOnPreferenceChangeListener().onPreferenceChange(this, true);
        }
    }
}

File: app/src/main/java/org/worshipsongs/service/ResetDefaultSettingsService.kt
package org.worshipsongs.service


import android.app.AlertDialog
import android.content.Context
import android.util.AttributeSet
import androidx.preference.DialogPreference
import androidx.preference.PreferenceManager
import org.worshipsongs.R

/**
 * @author Madasamy
 * @since  1.x
 */
class ResetDefaultSettingsService(var activityContext: Context, attrs: AttributeSet) :
        DialogPreference(activityContext, attrs)
{
    init
    {
        setTitle(R.string.reset_default_title)
    }

    override fun onClick()
    {
        val dialogBuilder = AlertDialog.Builder(context)
                .setTitle(context.getString(R.string.reset_default_title))
        dialogBuilder.setPositiveButton(R.string.ok) { dialog, which ->
            val preferencesEditor = PreferenceManager.getDefaultSharedPreferences(activityContext).edit()
            preferencesEditor.clear()
            PreferenceManager.setDefaultValues(activityContext, R.xml.settings, true)
            preferencesEditor.apply()
            onPreferenceChangeListener.onPreferenceChange(this, true)
        }
        dialogBuilder.setNegativeButton(R.string.cancel) { dialog, which ->
            dialog.cancel()
        }
        dialogBuilder.show()
    }
}
